### PR TITLE
add options to all Institution requests

### PR DESCRIPTION
**Pull request by tanatplaid**
change requires a version update. adds options to all institution methods.

### CODE REVIEW

File: plaid/institutions.go [hunk 1]
```diff
@@ -13,6 +13,32 @@ type Institution struct {
...
+	Logo string `json:"logo"`
+}
+
+type InstitutionStatus struct {
+	ItemLogins ItemLogins `json:"item_logins"`
+}
+
+type ItemLogins struct {
+	Status           string                     `json:"status"`
+	LastStatusChange string                     `json:"last_status_change"`
```
Comment (ntindall) [inline plaid/institutions.go]: should this be `time.Time` instead of a string? not sure how we serialize other timestamps.

Comment (tanatplaid) [reply plaid/institutions.go]: :zap:

File: plaid/institutions.go [hunk 2]
```diff
@@ -13,6 +14,32 @@ type Institution struct {
 	Name         string       `json:"name"`
 	Products     []string     `json:"products"`
 	CountryCodes []string     `json:"country_codes"`
+
+	// Included when `options.include_status` is true.
+	InstitutionStatus InstitutionStatus `json:"status"`
```
Comment (ntindall) [inline plaid/institutions.go]: if its nullable, should it be a pointer` ? I think thats the more idiomatic thing to do for structs.

Comment (tanatplaid) [reply plaid/institutions.go]: sure, but lmk know what you think after the update. primary colour and logo may be empty string which keeps as null anyway.

File: plaid/institutions.go [hunk 3]
```diff
@@ -45,9 +85,15 @@ type GetInstitutionByIDResponse struct {
 }
 
 type searchInstitutionsRequest struct {
-	Query     string   `json:"query"`
-	Products  []string `json:"products"`
-	PublicKey string   `json:"public_key"`
+	Query     string                    `json:"query"`
+	Products  []string                  `json:"products"`
+	PublicKey string                    `json:"public_key"`
+	Options   SearchInstitutionsOptions `json:"options",omitempty`
```
Comment (ntindall) [inline plaid/institutions.go]: this json struct tag is incorrect, should be `json:"options,omitempty"`

File: plaid/institutions.go [hunk 4]
```diff
@@ -13,6 +14,32 @@ type Institution struct {
 	Name         string       `json:"name"`
 	Products     []string     `json:"products"`
 	CountryCodes []string     `json:"country_codes"`
+
+	// Included when `options.include_status` is true.
+	InstitutionStatus *InstitutionStatus `json:"status",omitempty`
```
Comment (ntindall) [inline plaid/institutions.go]: omitempty needs to be inside of the `json` double quotes

**Pull request merged**